### PR TITLE
ci: add check for generated files

### PR DESCRIPTION
**Pull request by phm07**
This PR adds a CI check that ensures all generated files are present and up-to-date.
See #713

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,3 +47,23 @@ jobs:
...
+          go-version-file: go.mod
+
+      - name: Delete old generated files
+        run: find . -name 'zz_*.go' -delete
+
+      - name: Run go generate
+        run: go generate ./...
+
+      - name: Check uncommitted changes
+        run: git diff --exit-code
```
Comment (jooola) [inline .github/workflows/ci.yml:69]:
```suggestion
      - name: Check uncommitted changes
        run: git diff --exit-code
      
      - if: failure()
        run: echo "::error::Check failed, please run 'go generate ./...' and commit the changes."
```

File: internal/hcapi2/mock/mock_gen.go [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
-//go:generate mockgen -package hcapi2_mock -destination zz_network_client_mock.go github.com/hetznercloud/cli/internal/hcapi2 NetworkClient
-//go:generate mockgen -package hcapi2_mock -destination zz_server_client_mock.go github.com/hetznercloud/cli/internal/hcapi2 ServerClient
-//go:generate mockgen -package hcapi2_mock -destination zz_server_type_client_mock.go github.com/hetznercloud/cli/internal/hcapi2 ServerTypeClient
-//go:generate mockgen -package hcapi2_mock -destination zz_ssh_key_client_mock.go github.com/hetznercloud/cli/internal/hcapi2 SSHKeyClient
-//go:generate mockgen -package hcapi2_mock -destination zz_volume_client_mock.go github.com/hetznercloud/cli/internal/hcapi2 VolumeClient
-//go:generate mockgen -package hcapi2_mock -destination zz_placement_group_client_mock.go github.com/hetznercloud/cli/internal/hcapi2 PlacementGroupClient
-//go:generate mockgen -package hcapi2_mock -destination zz_rdns_client_mock.go github.com/hetznercloud/cli/internal/hcapi2 RDNSClient
+//go:generate go run github.com/golang/mock/mockgen -package hcapi2_mock -destination zz_action_client_mock.go github.com/hetznercloud/cli/internal/hcapi2 ActionClient
```
Comment (jooola) [inline internal/hcapi2/mock/mock_gen.go:3]: Does this go run respect the go.mod version ?

Comment (jooola) [reply internal/hcapi2/mock/mock_gen.go:3]:
```
If the package argument has a version suffix (like @latest or @v1.0.0),
"go run" builds the program in module-aware mode, ignoring the go.mod file in
the current directory or any parent directory, if there is one. This is useful
for running programs without affecting the dependencies of the main module.

If the package argument doesn't have a version suffix, "go run" may run in
module-aware mode or GOPATH mode, depending on the GO111MODULE environment
variable and the presence of a go.mod file. See 'go help modules' for details.
If module-aware mode is enabled, "go run" runs in the context of the main
module.

```

**Review by jooola — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -128,3 +128,9 @@ Additional Commands:
   set-rdns                    Change reverse DNS of a Server
   ssh                         Spawn an SSH connection for the server
 ```
+
+### Generated files
+
+Generated files (that are created by running `go generate`) should be prefixed with `zz_`. This is so that they are
+grouped together in the file list and are easy to identify as generated files. Also, it allows the CI to check if the
+generated files are up-to-date.
```
Comment (jooola) [inline CONTRIBUTING.md:136]:
```suggestion
Generated files (that are created by running `go generate`) should be prefixed with `zz_`. This is to group them together in the file list and to easily identify them as generated files. Also, it allows the CI to check if the
generated files are up-to-date.
```

**Review by apricote — APPROVED** [review]

**Pull request merged**